src/main/java/io/github/dsheirer/record/RecorderManager.java

```diff
@@ -225,7 +226,7 @@ private String getFilePrefix(ReusableAudioPacket packet)
             }
         }
 
-        return sb.toString();
+        return StringUtils.replaceIllegalCharacters(sb.toString());
```

Comment:
Could you please move the clean method to where each of the System/Site/Name values are added.  Otherwise, the clean method causes invalid linux filepaths.  

        sb.append(metadata.hasChannelConfigurationSystem() ? StringUtils.replaceIllegalCharacters(metadata.getChannelConfigurationSystem()) + "_" : "");
        sb.append(metadata.hasChannelConfigurationSite() ? StringUtils.replaceIllegalCharacters(metadata.getChannelConfigurationSite()) + "_" : "");
        sb.append(metadata.hasChannelConfigurationName() ? StringUtils.replaceIllegalCharacters(metadata.getChannelConfigurationName()) + "_" : "");

Reply:
I changed this suggestion a little bit to encompass the 'metadata.getPrimaryAddressTo().getIdentifier()'  calls as well. My thinking is that it may somehow be possible to have an illegal char in the identifier.